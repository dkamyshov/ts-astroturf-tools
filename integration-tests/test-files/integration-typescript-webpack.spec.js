const path = require('path');
const runner = require('./runner');
const fs = require('fs');
const testlist = require('../projects/integration-typescript-webpack/testlist');

jest.setTimeout(30000);

describe('typescript-webpack', () => {
  const workingDirectory = path.resolve(
    __dirname,
    '../projects/integration-typescript-webpack'
  );

  beforeAll(async () => {
    await runner.run(workingDirectory, 'yarn', []);
  });

  const executeTest = async (workingDirectory, testName) => {
    return runner.run(workingDirectory, 'yarn', [
      'webpack',
      `src/${testName}.tsx`,
      '--output-path',
      `dist/${testName}`,
    ]);
  };

  testlist.forEach(filename => {
    it(`css snapshot test - ${filename}`, async () => {
      const resultCSSFileContent = fs
        .readFileSync(
          path.resolve(workingDirectory, `dist/${filename}/style.css`)
        )
        .toString();

      expect(resultCSSFileContent).toMatchSnapshot();
    });
  });

  it(`shows missing CSS error for 'css'`, async () => {
    await expect(
      runner.rejectWithOutput(executeTest(workingDirectory, 'css-missing-css'))
    ).rejects.toContain(`Identifier "classB" is missing in corresponding CSS.`);
  });

  it(`shows unused CSS identifier warning for 'css'`, async () => {
    await expect(
      runner.resolveWithOutput(executeTest(workingDirectory, 'css-unused-css'))
    ).resolves.toContain(
      `Identifier "classB" is unused. Consider removing it from CSS.`
    );
  });
});
