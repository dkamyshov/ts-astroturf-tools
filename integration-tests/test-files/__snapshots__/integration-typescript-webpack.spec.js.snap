// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript-webpack css snapshot test - css 1`] = `
"
  .css-null-astroturf-local_classA {
    color: red;
  }

  .css-null-astroturf-local_classB {
    color: green;
  }


/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - css-local-interpolation 1`] = `
"
  .css-local-interpolation-null-astroturf-local_classA {
    color: red;
  }

  .css-local-interpolation-null-astroturf-local_classB {
    color: green;
  }


/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - styled 1`] = `
".styled-SomeComponent-astroturf-local_cls1 {
color: red;
}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - styled-local-interpolation 1`] = `
".styled-local-interpolation-SomeComponent-astroturf-local_cls1 {
color: red;
}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - styled-remote-interpolation 1`] = `
".styled-remote-interpolation-SomeComponent-astroturf-local_cls1 {
color: red;
background: green;
}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - xcss 1`] = `
"
.xcss-someClassName-astroturf-local_someClassName {

  color: red;

}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - xcss-local-interpolation 1`] = `
"
.xcss-local-interpolation-someClassName-astroturf-local_someClassName {

  color: red;

}


/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - xcss-property-assignment 1`] = `
"
.xcss-property-assignment-p_someClassName-astroturf-local_someClassName {

    color: red;
  
}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - xcss-property-assignment-local-interpolation 1`] = `
"
.xcss-property-assignment-local-interpolation-p_someClassName-astroturf-local_someClassName {

    color: red;
  
}


/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - xcss-property-assignment-remote-interpolation 1`] = `
"
.xcss-property-assignment-remote-interpolation-p_someClassName-astroturf-local_someClassName {

    color: red;
    background: green;
  
}


/*# sourceMappingURL=style.css.map*/"
`;

exports[`typescript-webpack css snapshot test - xcss-remote-interpolation 1`] = `
"
.xcss-remote-interpolation-someClassName-astroturf-local_someClassName {

  color: red;
  background: green;

}


/*# sourceMappingURL=style.css.map*/"
`;
